demos/classification_benchmark_demo/cpp_gapi/main.cpp

```diff
@@ -0,0 +1,313 @@
+        // clang-format off
+        const auto net =
+            cv::gapi::ie::Params<nets::Classification>{
+                FLAGS_m,  // path to topology IR
+                fileNameNoExt(FLAGS_m) + ".bin",  // path to weights
+                FLAGS_d  // device specifier
+            }.cfgNumRequests(config.maxAsyncRequests)
+             .pluginConfig(config.getLegacyConfig());
+        // clang-format on
+
+        auto pipeline = comp.compileStreaming(cv::compile_args(custom::kernels(), cv::gapi::networks(net)));
+
+        /** Output container for result **/
+        cv::Mat output;
+        IndexScore infer_result;
+
+        /** ---------------- The execution part ---------------- **/
+        std::shared_ptr<ImagesCapture> cap = openImagesCapture(FLAGS_i,
+                                                               true,
+                                                               read_type::safe,
+                                                               0,
+                                                               std::numeric_limits<size_t>::max(),
+                                                               stringToSize(FLAGS_res));
+        auto pipeline_inputs = cv::gin(cv::gapi::wip::make_src<custom::CommonCapSrc>(cap));
+        pipeline.setSource(std::move(pipeline_inputs));
+        std::string windowName = "Classification Benchmark demo G-API";
+        int delay = 1;
+
+        Presenter presenter(FLAGS_u, 0);
+        int width;
+        int height;
+        std::vector<std::string> gridMatRowsCols = split(FLAGS_res, 'x');
+        if (gridMatRowsCols.size() != 2) {
+            throw std::runtime_error("The value of ClassificationGridMat resolution flag is not valid.");
+        } else {
+            width = std::stoi(gridMatRowsCols[0]);
+            height = std::stoi(gridMatRowsCols[1]);
+        }
+
+        ClassificationGridMat gridMat(presenter, cv::Size(width, height));
+        bool keepRunning = true;
+        size_t framesNum = 0;
+        long long correctPredictionsCount = 0;
+        unsigned int framesNumOnCalculationStart = 0;
+        double accuracy = 0;
+        bool isTestMode = true;
+        std::chrono::steady_clock::duration elapsedSeconds = std::chrono::steady_clock::duration(0);
+        std::chrono::seconds testDuration = std::chrono::seconds(3);
+        std::chrono::seconds fpsCalculationDuration = std::chrono::seconds(1);
+        auto startTime = std::chrono::steady_clock::now();
+        pipeline.start();
+        IndexScore::LabelsStorage top_k_scored_labels;
+        top_k_scored_labels.reserve(FLAGS_nt);
+        int64_t timestamp = 0;
+        size_t total_produced_image_count = 0;
+        while (keepRunning && elapsedSeconds < std::chrono::seconds(FLAGS_time) && pipeline.pull(cv::gout(output, infer_result, timestamp))) {
+            std::chrono::microseconds dur(timestamp);
+            std::chrono::time_point<std::chrono::steady_clock> frame_timestamp(dur);
```

Comment:
Looks like this is the latency of the source itself

Reply:
To get the latency for pipeline iteration you need to measure time between `timestamp` that returned from the pipeline and `timestamp` after the `pull` is completed.

Reply:
this magic happens inside of `gridMat`
```
gridMat.textUpdate(metrics,
                               frame_timestamp,  <---- here
```
The logic of measure pipeline latency is quite similar to `classic benchmark demo`
1) mark source timestamp before invoke inferRequest
2) pass it through gridMat after pull